vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java

```diff
@@ -139,8 +145,34 @@ public void swipeToElement(SwipeDirection direction, Locator locator, Duration d
             });
         }
 
-        baseValidations.assertElementsNumber(String.format("The element by locator %s exists", locator), elements,
-                ComparisonRule.EQUAL_TO, 1);
+        if (baseValidations.assertElementsNumber(String.format("The element by locator %s exists", locator), elements,
+                ComparisonRule.EQUAL_TO, 1))
+        {
+            adjustElementPosition(elements.get(0), duration);
+        }
+    }
+
+    private void adjustElementPosition(WebElement element, Duration duration)
+    {
+        int windowSizeHeight = genericWebDriverManager.getSize().getHeight();
+        int windowCenterY = windowSizeHeight / 2;
+        int visibilityIndent = (int) (VISIBILITY_INDENT_COEFFICIENT * windowSizeHeight);
+        int elementTopCoordinateY = element.getLocation().getY();
+
+        if (elementTopCoordinateY < visibilityIndent)
+        {
+            performSwipe(windowCenterY, visibilityIndent - elementTopCoordinateY, duration);
+        }
+        else if (elementTopCoordinateY > windowSizeHeight - visibilityIndent)
+        {
+            performSwipe(windowCenterY, Math.negateExact(elementTopCoordinateY - (windowSizeHeight - visibilityIndent)),
+                    duration);
+        }
+    }
+
+    private void performSwipe(int centerY, int offset, Duration duration)
```

Comment:
what about moving this method to `TouchActions` and naming `performVerticalSwipe`